cql3/expr/prepare_expr.cc

```diff
@@ -986,13 +994,19 @@ static expression prepare_binop_lhs(const expression& lhs, data_dictionary::data
             }, sub.val);
 
             const abstract_type& sub_col_type = sub_col->column_specification->type->without_reversed();
-            if (!sub_col_type.is_map()) {
-                throw exceptions::invalid_request_exception(format("Column {} is not a map, cannot be used as a map", sub_col->name_as_text()));
+
+            lw_shared_ptr<column_specification> subscript_column_spec;
+            if (sub_col_type.is_map()) {
+                subscript_column_spec = map_key_spec_of(*sub_col->column_specification);
+            } else if (sub_col_type.is_list()) {
+                subscript_column_spec = list_key_spec_of(*sub_col->column_specification);
+            } else {
+                throw exceptions::invalid_request_exception(format("Column {} is not a map/list, cannot be subscripted", sub_col->name_as_text()));
```

Comment:
The error message is changed before the test that checks it. Because of it tests fail and it's not bisectable.

Reply:
Right, fixed.